56 update docs with regards to new storage and messaging config functionality (#67)
ethyca/fidesdocs: pages/fides/configuration/storage.md

| `details.auth_method` | The authentication method for creating a session with S3. Either `automatic` or `secret_keys`. |
| `details.max_retries` | The maximum number of retries to attempt when writing to the storage destination. |

##### Add credentials to your default S3 storage destination (if needed)

If your `auth_method` is set to `secret_keys`, you will need to also provide the following credential information for your storage destination via an additional API call:

```json filename="PUT /api/v1/storage/default/s3/secret"
  {
    "aws_access_key_id": "{{aws_access_key_id}}",
    "aws_secret_access_key": "{{aws_secret_access_key_id}}"
  }

```

See [additional credentials for S3 buckets](#additional-credentials-for-s3-buckets) section for details on these credentials.

##### Set `s3` as the active default storage destination

In order to make Fides actually use your default S3 storage configuration in privacy request execution, i.e. for all policy rules that do not have any other storage destination configured, you must make the following API call:

```json filename="PATCH /api/v1/config"
  {
    "storage": {
      "active_default_storage_type": "s3"
    }
  }
```

> <Callout> Out of the box, Fides assumes a `Local` storage destination as the active storage destination used, by default, for all policy rules that do not have ay storage destination configured. This is not recommended for production use cases, so users are encouraged to configure a default S3 storage destination, either through the UI or with the API calls described here. </Callout>

Reviewer: "Out of the box" and "by default" are repetitive, let's go with only one.

```suggestion
<Callout> Out of the box, Fides assumes a `Local` storage destination as the active storage destination used for all policy rules that do not have ay storage destination configured. This is not recommended for production use cases, so users are encouraged to configure a default S3 storage destination, either through the UI or with the API calls described here. </Callout>
```

> <Callout> Out of the box, Fides assumes a `Local` storage destination as the active storage destination used, by default, for all policy rules that do not have ay storage destination configured. This is not recommended for production use cases, so users are encouraged to configure a default S3 storage destination, either through the UI or with the API calls described here. </Callout>

Reviewer: Not recommended seems a bit too weak of a phrase for this. I think we _should_ use "should" instead.

```suggestion
<Callout> Out of the box, Fides assumes a `Local` storage destination as the active storage destination used, by default, for all policy rules that do not have ay storage destination configured. This should not be used for production use cases, so users should configure a default S3 storage destination, either through the UI or with the API calls described here. </Callout>
```

Author: 👍 for sure - good call here.

##### Check the default storage status

To check that the current active default storage destination is set to an S3 storage destination that's fully configured, you can invoke the following endpoint:

```json filename="GET /api/v1/storage/default/status"
{
    "config_status": "configured",
    "detail": "Active default storage location of type s3 is fully configured"
}
```

This can help you confirm that you've done what's needed to configure an S3 storage destination and that Fides will use it as the active default.

#### Configure _manual_ (non-default) storage destinations

> If you want even finer grained control and do not want to only rely on a _default_ storage configuration, you can also configure _manual_ storage configurations and explicitly associate them with given policy rules. This allows you to output particular types of requested data to particular storage destinations, for example.

Reviewer: https://en.wiktionary.org/wiki/fine-grained#:~:text=fine-grained%20(comparative%20more%20fine,(figurative)%20Highly%20detailed.

```suggestion
If you want even more fine-grained control and do not want to only rely on a _default_ storage configuration, you can also configure _manual_ storage configurations and explicitly associate them with given policy rules. This allows you to output particular types of requested data to particular storage destinations, for example.
```

Author: :)

Author: how about "less coarse grained" :trollface:
